index.go

```diff
@@ -20,9 +37,18 @@ func rootHandler(
 }
 
 func main() {
+    addr, err := determineListenAddress()
+    if err != nil {
+        log.Fatal(err)
+    }
+
     // set handler
     http.HandleFunc("/", rootHandler)
 
     // start server
-    log.Fatal(http.ListenAndServe(":8080", nil))
+    log.Printf("Listening on %s...\n", addr)
+    err := http.ListenAndServe(addr, nil)
```

Comment:
I think this was better when it was part of the if statement

Reply:
will change to if block as discussed.